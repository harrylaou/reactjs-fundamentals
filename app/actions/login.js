export const LOGIN_USER_OK = 'LOGIN_USER_OK'
export const LOGOUT_USER = 'LOGOUT_USER'

export const loginUser = () => ({
  type: LOGIN_USER_OK
})

export const logoutUser = () => ({
  type: LOGOUT_USER
})