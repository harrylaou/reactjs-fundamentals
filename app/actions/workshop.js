export const RECEIVE_WORKSHOP = 'RECEIVE_WORKSHOP'


export const receiveWorkshop = workshop => ({
  type: RECEIVE_WORKSHOP,
  workshop
})