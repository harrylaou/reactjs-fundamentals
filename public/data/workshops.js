[
  {
    "title": "React Router and fetch data",
    "url": "http://www.meetup.com/JS-Lab-London/events/231907561/",
    "students" : "Unlimited",
    "price": "Free",
    "instructors": [
        {
            "name":"Alex Lobera",
            "url": "http://alexlobera.com/",
            "avatar": "https://pbs.twimg.com/profile_images/1540939151/whereIsAlexWithGlasses.jpg"
        }
    ]
  },
  {
    "title": "React Master class",
    "url": "http://techdency.com/#react-training",
    "price": "Â£ 595",
    "students" : "10 per instructor",
    "instructors": [
        {
            "name":"Alex Lobera",
            "url": "http://alexlobera.com/",
            "avatar": "https://pbs.twimg.com/profile_images/1540939151/whereIsAlexWithGlasses.jpg"
        },
        {
            "name":"Davy Engone",
            "url": "https://github.com/davyengone",
            "avatar": "https://pbs.twimg.com/profile_images/589582545022545921/qQRHggUr.jpg"
        }
    ]
  }
]