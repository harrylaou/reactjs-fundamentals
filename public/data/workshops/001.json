{
  "id": "001",
  "title": "React Router and fetch data",
  "url": "http://www.meetup.com/JS-Lab-London/events/231907561/",
  "students" : "Unlimited",
  "price": "Free",
  "instructors": [
    {
      "name":"Alex Lobera",
      "url": "http://alexlobera.com/",
      "avatar": "https://pbs.twimg.com/profile_images/1540939151/whereIsAlexWithGlasses_400x400.jpg"
    }
  ]
}