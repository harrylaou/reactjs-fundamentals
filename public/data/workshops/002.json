{
  "id": "002",
  "title": "React Master class",
  "url": "http://techdency.com/#react-training",
  "price": "Â£ 595",
  "students" : "10 per instructor",
  "instructors": [
    {
      "name":"Alex Lobera",
      "url": "http://alexlobera.com/",
      "avatar": "https://pbs.twimg.com/profile_images/1540939151/whereIsAlexWithGlasses_400x400.jpg"
    },
    {
      "name":"Mark Rivelles",
      "url": "https://github.com/fsdevlondon",
      "avatar": "https://pbs.twimg.com/profile_images/784142672370003969/Hj51EXIt.jpg"
    }
  ]
}