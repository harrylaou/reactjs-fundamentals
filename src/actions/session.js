export const LOG_OUT = 'LOG_OUT'
export const LOGGED_IN = 'LOGGED_IN'

export const logout = () => ({
  type: LOG_OUT
})

export const loggedIn = token => ({
  type: LOGGED_IN,
  token
})
