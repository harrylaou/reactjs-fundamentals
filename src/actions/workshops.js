export const RECEIVE_WORKSHOPS = 'RECEIVE_WORKSHOPS'

export const receiveWorkshops = workshops => ({
  type: RECEIVE_WORKSHOPS,
  workshops
})
