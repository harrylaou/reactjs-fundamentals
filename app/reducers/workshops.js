import {RECEIVE_WORKSHOPS} from '../actions/workshops'

const workshops = (state = [], action) => {
  switch (action.type) {
    case RECEIVE_WORKSHOPS: {
      return action.workshops
    }
    default:
      return state
  }
}

export default workshops
