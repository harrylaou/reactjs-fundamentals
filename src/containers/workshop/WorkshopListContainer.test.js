import React from 'react'
import { shallow, mount } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import ConnectedWorkshopListContainer, {WorkshopListContainer} from './WorkshopListContainer'
import { mockFetchContainer } from '../../testUtils'
import getMuiTheme from 'material-ui/styles/getMuiTheme'

import FakeConfigureStore from 'redux-mock-store'
import ActualConfigureStore from '../../store'
import {Provider} from 'react-redux'
import * as actions from '../../actions/workshops'

import injectTapEventPlugin from 'react-tap-event-plugin'
injectTapEventPlugin()

describe('<WorkshopListContainer />', () => {
  it('should render with default props', () => {
    const push = jest.fn()
    const props = {
      params: { workshop: '123' },
      width: 3,
      router: {push},
      workshops: []
    }

    const wrapper = shallow(
      <WorkshopListContainer {...props} />
    )

    expect(shallowToJson(wrapper)).toMatchSnapshot()

    wrapper.instance().showWorkshop({id: 'reactlondon'})

    expect(push).toBeCalledWith('/workshops/reactlondon')
  })


  const workshopData = {
    workshops: [{
      title: 'reactjs_fundamentals',
      price: 990,
      url: 'https://reactjs.academy/react-redux-training-london',
      instructors: [{
        name: 'Manolo',
        url: '',
        avatar: ''
      }]
    }]
  }

  const initialState = {
    workshops: [{
      title: 'reactjs_advanced',
      price: 990,
      url: 'https://reactjs.academy/react-redux-training-london',
      instructors: [{
        name: 'Paco',
        url: '',
        avatar: ''
      }]
    }]
  }
  const mockStore = FakeConfigureStore()

  const push = jest.fn()
  const replace = () => {}
  const go = () => {}
  const goBack = () => {}
  const goForward = () => {}
  const setRouteLeaveHook = () => {}
  const isActive = () => {}

  const props = {
    params: { workshop: '123' },
    width: 3,
    router: {
      push,
      replace,
      go,
      goBack,
      goForward,
      setRouteLeaveHook,
      isActive,
      setRouteLeaveHook
    },
    workshops: []
  }
  let store

  describe('<ConnectedWorkshopListContainer />', () => {
    it('should render connected component', () => {
      store = mockStore(initialState)
      const container = shallow(
        <ConnectedWorkshopListContainer
          {...props}
          store={store}
        />
      )

      expect(shallowToJson(container)).toMatchSnapshot()
    })
  })

  describe('Integration test: Mount + Wrapping Provider', () => {
    let wrapper

    beforeEach(()=>{
      store = mockStore(initialState)

      mockFetchContainer(workshopData.workshops)

      wrapper = mount(
        <Provider store={store} >
          <ConnectedWorkshopListContainer
            {...props}
          />
        </Provider>,
        {
          context: {
            muiTheme: getMuiTheme()
          },
          childContextTypes:{
            muiTheme: React.PropTypes.object.isRequired,
          }
        }
      )
    })

    it('should mount and map connect connected component', () => {
      expect(wrapper.find(ConnectedWorkshopListContainer).length).toEqual(1)
    })

    it('should Prop matches with initialState', () => {
      expect(wrapper.find(WorkshopListContainer).prop('workshops')).toEqual(initialState.workshops)
    })

    it('should check action dispatch', () => {
      store.dispatch(actions.receiveWorkshops(workshopData.workshops))
      const action = store.getActions()

      expect(action[0].type).toBe(actions.RECEIVE_WORKSHOPS)
    })
  })

  describe('Integration test: Mount + Wrapping Provider + Actual Store + Reducers', () => {
    let wrapper

    beforeEach(()=>{
      store = ActualConfigureStore(initialState)

      mockFetchContainer(workshopData.workshops)

      wrapper = mount(
        <Provider store={store} >
          <ConnectedWorkshopListContainer
            {...props}
          />
        </Provider>,
        {
          context: {
            muiTheme: getMuiTheme()
          },
          childContextTypes:{
            muiTheme: React.PropTypes.object.isRequired,
          }
        }
      )
    })

    it('should mount and map connect connected component', () => {
      expect(wrapper.find(ConnectedWorkshopListContainer).length).toEqual(1)
    })

    it('should check action dispatch', () => {
      //OR store.dispatch(actions.receiveWorkshops(workshopData))
      wrapper.find(WorkshopListContainer).props().dispatch(actions.receiveWorkshops(workshopData.workshops))

      expect(wrapper.find(WorkshopListContainer).prop('workshops')).toEqual(workshopData.workshops)
    })
  })
})