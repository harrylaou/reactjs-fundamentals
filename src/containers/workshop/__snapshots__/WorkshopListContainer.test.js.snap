// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WorkshopListContainer /> <ConnectedWorkshopListContainer /> should render connected component 1`] = `
<withRouter(WithWidth)
  dispatch={[Function]}
  params={
    Object {
      "workshop": "123",
    }
  }
  router={
    Object {
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "isActive": [Function],
      "push": [Function],
      "replace": [Function],
      "setRouteLeaveHook": [Function],
    }
  }
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "notify": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": null,
    }
  }
  width={3}
  workshops={
    Array [
      Object {
        "instructors": Array [
          Object {
            "avatar": "",
            "name": "Paco",
            "url": "",
          },
        ],
        "price": 990,
        "title": "reactjs_advanced",
        "url": "https://reactjs.academy/react-redux-training-london",
      },
    ]
  }
/>
`;

exports[`<WorkshopListContainer /> should render with default props 1`] = `
<WorkshopList
  params={
    Object {
      "workshop": "123",
    }
  }
  router={
    Object {
      "push": [Function],
    }
  }
  showWorkshop={[Function]}
  width={3}
  workshop="123"
  workshops={Array []}
/>
`;
